File: cpp/ulamek/ulamek.cpp
/*
 * ulamek.cpp
 * 
 */


#include <iostream>
#include <cmath>
#include "ulamek.h"
#include <stdlib.h>
using namespace std;

bool Ulamek::set_l(int li){
	l = li;
	return true;
	
}
bool Ulamek::set_m(int mi){
	if (mi!= 0) {
		m = mi;
		return true;
	}
	else {
		cout << "Mianownik jest rowny zero" << endl;
		return false;
    }
}
bool Ulamek::set_lm(int li, int mi){
	int q=skroc(li, mi);
	l = li/q;
	return set_m(mi/q);
	}
void Ulamek::drukuj() {
    cout << l << "/" << m;
}
Ulamek::Ulamek(int li, int mi)
{ 
	int q=skroc(li, mi);
	if(mi < 0) q = -q;
	l = li/q;
	m = mi/q;
}

int Ulamek::skroc(int li, int mi)
	{ int i=0;
	while (li > 0){
		i++;
		li = li%mi;
		mi = mi-li;
	}
		return mi;
	}

File: python/ulamek/ulamek.py
class Ulamek:
	def __init__(self, l=0, m=1):
		self.l = int(l)
		self.m = int(m)
		self.__ustawMian(m)
		self.setLicz(l)
		self.setMian(m)
		self.getLicz()
		self.getMian()
	def __ustawMian(self, m):
		if (m != 0):
			self.m = m
			return True
		else:
			print("Mianownik jest równy zero!")
			return False
	def drukuj(self):
		print(self.l, "/", self.m)
	def setLicz(self, l):
		self.l = l
		return True
	def setMian(self, m):
		if(m != 0):
			self.m = m
			return True
		else:
			print("Mianownik jest równy zero")
			return False
	def getLicz(self):
		return self.l
	def getMian(self):
		return self.m
#def nwdv1(a, b):
#    """ wersja klasyczna """
#    while a != b:
 #       if a > b:
   #         a = a - b
   #     else: 
    #        b = b - a
    # print(a)
   # return a

